bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -126,32 +122,47 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     updateState(OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_CENTRAL_HEATING2_ENABLED,
                             OnOffType.from(!gatewayCommand.getMessage().equals("0.0")));
                 }
-            } else {
-                connect();
             }
         }
     }
 
     @Override
     public void connecting() {
+        // decouple threads via a scheduler task
+        scheduler.submit(() -> connectingTask());
```

Comment:
Why does this need to be done in separate thread?

Reply:
I found that the callback tasks, which access the framework, sometimes block; which means that the polling thread gets delayed. And by firing a scheduler.submit task, it can completely be decoupled.

Reply:
I missed that it was a callback, thanks.